pil-analyzer/src/pil_analyzer.rs

```diff
@@ -144,8 +152,21 @@ impl PILAnalyzer {
 
         for PILFile(file) in files {
             self.current_namespace = Default::default();
-            for statement in file {
-                self.handle_statement(statement);
+            for ref statement in file {
+                if let PilStatement::TraitImplementation(_, trait_impl) = statement {
```

Comment:
I don't see why trait implementations cannot be handled in the same way as other pil statements:

StatementProcessor::handle_statement gets another section for trait implementations and returns a new kind of PILItem. There you also have access to the AnalysisDriver and the expression processor to turn parsed types into analyzed types.